lib/kubernetes-deploy/kubernetes_resource/cloudsql.rb

```diff
@@ -0,0 +1,66 @@
+module KubernetesDeploy
+  class Cloudsql < KubernetesResource
+    TIMEOUT = 30.seconds
+
+    def initialize(name, namespace, file)
+      @name, @namespace, @file = name, namespace, file
+    end
+
+    def sync
+      _, st = run_kubectl("get", type, @name)
+      @found = st.success?
+      @status = true
+
+
+
+      log_status
+    end
+
+    def deploy_succeeded?
+      cloudsql_proxy_deployment_exists? && mysql_service_exists?
+    end
+
+    def deploy_failed?
+      !cloudsql_proxy_deployment_exists? || !mysql_service_exists?
+    end
+
+    def tpr?
+      true
+    end
+
+    def exists?
+      @found
+    end
+
+    private
+    def cloudsql_proxy_deployment_exists?
+      deployments, st = run_kubectl("get", "deployments", "--selector", "name=cloudsql-proxy", "--namespace=#{@namespace}", "-o=json")
```

Comment:
Since you're only interested in one, why not use `"get", "deployments, "cloudsql-proxy", "--output=json"`? Note that you don't need to send the namespace. It is always required in the context of these classes, so `run_kubectl` adds it and in fact raises if `@namespace` is blank.

(same questions/comments about the mysql version below)